[v2] Port logic to prompt-toolkit 3

kyleknap opened the pull request:
Fixes https://github.com/aws/aws-cli/issues/6507

The main difference is that PTK3 uses the built-in Python asyncio event loop. This required a large refactor of the test scaffolding as some of the synchronous assumptions were no longer valid. With regards to refactoring, I pivoted it to spinning up the application in a thread and then feed input keys to the app and make assertions on it while the app was running. This was also cleaner in that we did not have to add actions and assertions to a queue to be processed and instead can just make the assertion/action directly.

stealthycoin commented on tests/unit/customizations/wizard/test_app.py:
```diff
@@ -36,12 +31,482 @@
 from awscli.customizations.wizard.core import BaseStep, Executor
 
 
+@pytest.fixture
```
How slow is all this stuff? I really don't want the unit tests to be significantly slowed down by these tests. It seems more reasonable for them to be functional tests.

kyleknap replied:
In the next commit, it should be much faster as there are not blanket sleep statements

kyleknap commented on tests/__init__.py:
```diff
@@ -370,117 +374,82 @@ def stream(self, **kwargs):
-        _stdout = TextIOWrapper(BytesIO(), encoding="utf-8")
-        self._app.output.stdout = _stdout
-        self._app.run(pre_run=pre_run)
+            self._wait_until_app_ui_is_no_longer_redrawing()
```
I ended up going back to that in the next commit

kyleknap commented:
@stealthycoin I made some adjustments to how we wait for the app updates to propagate and also incorporated your other feedback. I ended up using the `after_render` property to help drive when to stop waiting. The tests run much faster now and I believe I removed any of the flakiness from them.
